### DISCUSSION TITLE
Localized grain refinement

### DISCUSSION THREAD

**Discussion by ccis88** [neperfepx/neper#664] [Q&A]
I want to refine the grain in the inner part of a cylinder, how should I do it?
I have attached schematics to show a 2D interface instead of a 3D one. The red area is the cube area that I want to refine.
Please help me. Thank you.
![5be89d658cd5921717b7291705dbe8b](https://user-images.githubusercontent.com/122538253/226109079-444a82f7-c558-4056-b23a-455afac78652.jpg)

**Comment by rquey** [comment]
The only way I see is by a 2-scale tessellation:
```
neper -T -n "50::((id==1)?50:1)" -domain "circle(1,100)" -dim 2  -o d
neper -V d.tess -print img
```
![image](https://user-images.githubusercontent.com/2659376/227145188-d824244a-8135-4d5b-878d-96b24af698fe.png)

**Reply by ccis88** [reply]
Thanks for the answer, I guess this can be fixed by setting seed points.

**Reply by ccis88** [reply]
Hello, I get the following error when making grains via seed points. My neperb version is 4.5.1-4 and I would like to ask what is the cause
```
========================    N   e   p   e   r    =======================
Info   : A software package for polycrystal generation and meshing.
Info   : Version 4.5.1-4
Info   : Built with: gsl|muparser|opengjk|openmp|nlopt|libscotch (full)
Info   : Running on 8 threads.
Info   : <https://neper.info>
Info   : Copyright (C) 2003-2022, and GNU GPL'd, by Romain Quey.
Info   : No initialization file found (`/home/cc1998/.neperrc').
Info   : ---------------------------------------------------------------
Info   : MODULE  -T loaded with arguments:
Info   : [ini file] (none)
Info   : [com line] -n 2290170 -domain
         cylinder(0.1,18):translate(-9,-9,0) -morphooptiini
         coo:file(coo),weight:0 -morpho voronoi -transform
         cut(hspace(0,-1,0,0)) -regularization 1
Info   : ---------------------------------------------------------------
Info   : Reading input data...
Info   : Creating domain...
Info   : Creating tessellation...
Info   :   - Setting seeds... Segmentation fault (core dumped)
```

**Reply by rquey** [reply]
Yes, it can also be done somehow by loading custom seeds.

I cannot say about your error without the coo file, but I notice that the number of seeds is very large...

**Reply by ccis88** [reply]
Yes, the number of seeds is large, but it is possible to succeed with the number of seeds and the model scaled down in the same proportion. So I'm wondering if my computer equipment performance b not enough
